code-editor/auto-stop-idle/on-start.sh

```diff
@@ -17,8 +17,25 @@ PYTHON_SCRIPT_PATH=$SOLUTION_DIR/sagemaker_code_editor_auto_shut_down/auto_stop_
 
 # Installing cron
 sudo apt-get update -y
-sudo sh -c 'printf "#!/bin/sh\nexit 0" > /usr/sbin/policy-rc.d'
-sudo apt-get install -y cron
+
+# Issue - https://github.com/aws-samples/sagemaker-studio-apps-lifecycle-config-examples/issues/12
+# SM Distribution image 1.6 is not starting cron service by default https://github.com/aws/sagemaker-distribution/issues/354
+
+# Check if cron needs to be installed  ## Handle scenario where script exiting("set -eux") due to non-zero return code by adding true command.
+status="$(dpkg-query -W --showformat='${db:Status-Status}' "cron" 2>&1)" || true 
+if [ ! $? = 0 ] || [ ! "$status" = installed ]; then
+	# Fixing invoke-rc.d: policy-rc.d denied execution of restart.
+	sudo /bin/bash -c "echo '#!/bin/sh
+	exit 0' > /usr/sbin/policy-rc.d"
+
+	# Installing cron.
+	echo "Installing cron..."
+	sudo apt install cron
```

Comment:
Seems this is not working if it's running in a private vpc(no internet access). Can we add mention this in readme file? 
This is not a new issue from this PR.

Reply:
Added [steps](https://github.com/lavaraja/sagemaker-studio-apps-lifecycle-config-examples/tree/main/code-editor/auto-stop-idle#steps-for-installing-in-internet-free-vpc-environments) for installing in in a private vpc(no internet access) and also additional steps for sm distribution image version 1.5 and below(where cron package was not part of the image).